src/index.ts

```diff
@@ -165,7 +170,13 @@ export class Cinnamon {
                 ) &&
                 !token
             ) {
-                await this.refreshLogin({ refreshToken: this.refreshToken });
+                if (!this.refreshTokenRequest) {
+                    this.refreshTokenRequest = this.refreshLogin({
```

Comment:
This is missing an `await` before `this.refreshLogin`? Otherwise it will not cleanup itself because `Promise<null>` is coerced to true.

Reply:
no thats what should happen.
all calls should wait for this call, we want to promise

Reply:
No actually I think you need to do:

```javascript
this.refreshTokenRequest = this.refreshLogin({ ... });
this.refreshTokenRequest = await this.refreshTokenRequest;
```